src/IndexRegistry.sol

```diff
@@ -117,121 +126,170 @@ contract IndexRegistry is IndexRegistryStorage {
     *******************************************************************************/
 
     /**
-     * @notice updates the total numbers of operator in `quorumNumber` to `numOperators`
-     * @param quorumNumber is the number of the quorum to update
-     * @param numOperators is the number of operators in the quorum
+     * @notice Increases the historical operator count by 1 and returns the new count
      */
-    function _updateTotalOperatorHistory(uint8 quorumNumber, uint32 numOperators) internal {
-        _totalOperatorsHistory[quorumNumber].push(QuorumUpdate({
-            numOperators: numOperators,
-            fromBlockNumber: uint32(block.number)
-        }));
+    function _increaseOperatorCount(uint8 quorumNumber) internal returns (uint32) {
+        QuorumUpdate storage lastUpdate = _latestQuorumUpdate(quorumNumber);
+        uint32 newOperatorCount = lastUpdate.numOperators + 1;
+        
+        _updateOperatorCountHistory(quorumNumber, lastUpdate, newOperatorCount);
+
+        // If this is the first time we're using this index, push its first update
+        // This maintains an invariant: existing indices have nonzero history
+        if (_indexHistory[quorumNumber][newOperatorCount - 1].length == 0) {
+            _indexHistory[quorumNumber][newOperatorCount - 1].push(OperatorUpdate({
+                operatorId: OPERATOR_DOES_NOT_EXIST_ID,
+                fromBlockNumber: uint32(block.number)
+            }));
+        }
```

Comment:
seems somewhat odd to push this here with the `OPERATOR_DOES_NOT_EXIST_ID` id?
I'll keep looking and try to circle back to this though

Reply:
Ah, I see, it's necessary so that the history has nonzero length when we then call `_assignOperatorToIndex` immediately afterward.
Works for me.
I guess the property here to ensure we're using these consistently is (at minimum)
```
if (_indexHistory[quorumNumber][newOperatorCount - 1].length == 1) {
_indexHistory[quorumNumber][newOperatorCount - 1][0].operatorId != 0;
}
```
i.e. the end result should always be that the first update *ends up* with a nonzero `operatorId`, even though we set it to zero here

Reply:
yep. i'm not super happy with the way i added this to the increase count method, but couldn't find a different way that kept things super simple. ill think about it this morning

Reply:
i'm going to keep this for now, but we can revisit when the PRs are merged :)